ingest/expression_files/expression_files.py

```diff
@@ -64,6 +64,41 @@ def set_dataArray(self):
         Each expression file will have its own implementation of setting the
         DataArray with expression data."""
 
+    @staticmethod
+    def has_unique_cells(cell_names: List, study_id, client):
+        """Checks cell names against database to confirm matrix contains unique
+            cell names
+
+         Parameters:
+            cell_names (List[str]): List of cell names in matrix
+            study_id (ObjectId): The study id the cell names belong to
+            client : MongoDB client
+        """
+        COLLECTION_NAME = "data_arrays"
+        query = {
+            "$and": [
+                {"linear_data_type": "Study"},
+                {"array_type": "cells"},
+                {"study_id": study_id},
+            ]
+        }
+        # Returned fields from query results
+        field_names = {"values": 1, "_id": 0}
+        # Dict = {values_1: [<cell names>]... values_n:[<cell names>]}
+        query_results: List[Dict] = list(
+            client[COLLECTION_NAME].find(query, field_names)
+        )
+        # Query did not return results
+        if not query_results:
+            return True
+        # Flatten query results
+        existing_cells = [
+            values
+            for cell_values in query_results
+            for values in cell_values.get("values")
+        ]
+        return not any(name in existing_cells for name in cell_names)
```

Comment:
I would use set intersection here so you can identify the total number and first few duplicates. set(lst1) & set(lst2)

Reply:
I believe you took care of this